src/interface/manager.ts

```diff
@@ -7,9 +7,31 @@ import { Meta } from './meta';
 export abstract class Manager {
 
     protected name: string;
+    private cached: { [key: string]: any };
 
     constructor(name: string) {
         this.name = name;
+        this.cached = {};
+    }
+
+    private requestCache(key: string, value?: any) : any
+    {
+        if (!value) {
+            return this.cached[key];
+        }
+        this.cached[key] = value;
+    }
+
+    protected async cache(
+        request: (name: string, version?: string) => Promise<any>, 
+        name: string, 
+        version?: string ) : Promise<any> 
+        {
+        const key = `${name}:${version}`;
+        if(!this.requestCache(key)) {
```

Comment:
Ha! It's possible that you will start several promises before 1st result will be saved.